packages/common/src/services/ApiService.ts

```diff
@@ -44,20 +45,64 @@ export default class ApiService {
     return date ? parseISO(date) : undefined;
   };
 
+  /**
+   * Transform incoming content lists
+   */
+  private transformContentList = (contentList: ContentList): Playlist => {
+    const { list, ...rest } = contentList;
+
+    const playlist: Playlist = { ...rest, playlist: [] };
+
+    playlist.playlist = list.map((item) => {
+      const { custom_params, media_id, description, tags, ...rest } = item;
+
+      const playlistItem: PlaylistItem = {
+        feedid: contentList.id,
+        mediaid: media_id,
+        tags: tags.join(','),
+        description: description || '',
+        sources: [],
+        images: [],
+        image: '',
+        link: '',
+        pubdate: 0,
+        ...rest,
+        ...custom_params,
+      };
+
+      return this.transformMediaItem(playlistItem, playlist);
+    });
+
+    return playlist;
+  };
+
+  /**
+   * Transform incoming playlists
+   */
+  private transformPlaylist = (playlist: Playlist, relatedMediaId?: string) => {
+    playlist.playlist = playlist.playlist.map((item) => this.transformMediaItem(item, playlist));
+
+    // remove the related media item (when this is a recommendations playlist)
+    if (relatedMediaId) playlist.playlist.filter((item) => item.mediaid !== relatedMediaId);
```

Comment:
https://www.imperialviolet.org/2014/02/22/applebug.html
```suggestion
    if (relatedMediaId) {
      playlist.playlist.filter((item) => item.mediaid !== relatedMediaId);
    }
```

Reply:
Hm, looks like this is the way it worked before. I will assign it.

Reply:
Probably just a bug then.. good to fix while we're here